stat_test.go

```diff
@@ -836,33 +837,89 @@ var linearRegressionTests = []struct {
 
 		tol: 1e-5,
 	},
+	{
+		name: "x, y length mismatch",
+
+		x:       make([]float64, 3),
+		y:       make([]float64, 2),
+		weights: make([]float64, 3),
+	},
+	{
+		name: "weights mismatch with x",
+
+		x:       make([]float64, 3),
+		y:       make([]float64, 3),
+		weights: make([]float64, 2),
+	},
+	{
+		name: "x, y length mismatch and weights nil",
+
+		x:       make([]float64, 3),
+		y:       make([]float64, 2),
+		weights: nil,
+	},
+	{
+		name:    "weights are nil",
+		x:       []float64{12.0, 24.0, 21.0},
+		y:       []float64{1.0, 2.0, 3.0},
+		weights: nil,
+	},
 }
 
 func TestLinearRegression(t *testing.T) {
 	for _, test := range linearRegressionTests {
-		alpha, beta := LinearRegression(test.x, test.y, test.weights, test.origin)
-		var r float64
-		if test.origin {
-			r = RNoughtSquared(test.x, test.y, test.weights, beta)
+		if strings.Contains(test.name, "mismatch") {
+			if !Panics(func() { LinearRegression(test.x, test.y, test.weights, true) }) {
+				t.Errorf("LinearRegression did not panic with length mismatch")
+			}
+			if !Panics(func() { LinearRegression(test.x, test.y, test.weights, false) }) {
+				t.Errorf("LinearRegression did not panic with length mismatch")
+			}
+			if !Panics(func() { RSquared(test.x, test.y, test.weights, 0, 0) }) {
+				t.Errorf("RSquared did not panic with lengths length mismatch")
+			}
+			if !Panics(func() { RSquaredFrom(test.x, test.y, test.weights) }) {
+				t.Errorf("RSquaredFrom did not panic with weights length mismatch")
+			}
+			if !Panics(func() { RNoughtSquared(test.x, test.y, test.weights, 0) }) {
+				t.Errorf("RNoughtSquared did not panic with weights length mismatch")
+			}
+		} else if test.name == "weights are nil" {
```

Comment:
Remove this else if. As above, I don't like the string match, and hand-coding the weight size is not good practice. The test isn't bad, but it should be put after the checks below. At that point you can do

if test.weights != nil { 
    continue
}
weights = make([]float64, len(test.x))
 for i := range weights {
      weights[i] = 1
}
````
Then make the calls and check that the sizes match.